libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -309,6 +311,18 @@ assembly::VariableDeclaration Parser::parseVariableDeclaration()
 	return varDecl;
 }
 
+assembly::ForLoop Parser:parseForLoop()
+{
+	ForLoop forLoop = createWithLocation<ForLoop>();
+	expectToken(Token::For);
+	forLoop.preCondition = parseBlock();
```

Comment:
It is not really a "condition", the only condition is what is now called "expression".
What about:
`init`, `condition`, `update` (I have seen that in other c-like grammars)